src/framework/uicomponents/qml/MuseScore/UiComponents/internal/GridViewDelegate.qml

```diff
@@ -41,14 +41,17 @@ Item {
     GridView {
         id: gridView
 
-        property int columns: root.columns !== -1 ? root.columns : gridView.count
+        property int columns: root.columns !== -1 ? root.columns : Math.ceil(gridView.count / gridView.rows)
         property int rows: root.rows !== -1 ? root.rows : Math.ceil(gridView.count / gridView.columns)
 
+        x: -1/2 * root.columnSpacing
```

Comment:
I think you can replace this on anchors with margins

Reply:
Currently that is not possible, because `root.width` depends on `gridview.width`. 
We could do the following: 
It is more verbose, but maybe also "feels" more correct. Do you think it's worth it?
```
Item {
    id: root
    ...
    width: gridView.columns * gridView.cellWidth - root.columnSpacing
    height: gridView.rows * gridView.cellHeight - root.rowSpacing

    GridView {
        id: gridView
        ...
        anchors.fill: parent
        anchors.topMargin: -root.rowSpacing / 2
        anchors.bottomMargin: -root.rowSpacing / 2
        anchors.leftMargin: -root.columnSpacing / 2
        anchors.rightMargin: -root.columnSpacing / 2

        cellWidth: ...
    }
}